packages/patternfly-4/react-core/src/components/Popover/Popover.js

```diff
@@ -1,40 +1,300 @@
+    this.hideOrNotify();
+  };
+
+  hideAllPopovers = () => {
+    document.querySelectorAll('.tippy-popper').forEach(popper => {
+      popper._tippy && popper._tippy.hide();
+    });
+  };
+
+  onHide = tip => {
+    this.state.isOpen && this.setState({ isOpen: false });
+    return this.props.onHide(tip);
+  };
+
+  onHidden = tip => this.props.onHidden(tip);
+
+  onMount = tip => this.props.onMount(tip);
+
+  onShow = tip => {
+    const { hideOnOutsideClick, isVisible, onShow } = this.props;
+    // hide all other open popovers first if events are managed by us
+    !hideOnOutsideClick && isVisible === null && this.hideAllPopovers();
+    this.state.isOpen === false && this.setState({ isOpen: true });
+    return onShow(tip);
+  };
+
+  onShown = tip => this.props.onShown(tip);
+
+  render() {
+    const {
+      position,
+      enableFlip,
+      children,
+      className,
+      'aria-label': ariaLabel,
+      headerContent,
+      bodyContent,
+      isVisible,
+      shouldClose,
+      appendTo,
+      hideOnOutsideClick,
+      onHide,
+      onHidden,
+      onShow,
+      onShown,
+      onMount,
+      zIndex,
+      size,
+      ...rest
+    } = this.props;
+    const content = (
+      <GenerateId>
+        {randomId =>
+          this.state.isOpen && (
+            <FocusTrap focusTrapOptions={{ clickOutsideDeactivates: true }}>
+              <div
+                className={css(
+                  styles.popover,
+                  !enableFlip && getModifier(styles, position, styles.modifiers.top),
+                  className
+                )}
+                role="dialog"
+                aria-modal="true"
+                aria-label={headerContent ? undefined : ariaLabel}
+                aria-labelledby={headerContent ? `popover-${randomId}-header` : undefined}
+                aria-describedby={`popover-${randomId}-body`}
+                {...rest}
+              >
+                <PopoverArrow />
+                <PopoverContent>
+                  <PopoverCloseButton onClose={this.closePopover} />
+                  {headerContent && <PopoverHeader id={`popover-${randomId}-header`}>{headerContent}</PopoverHeader>}
+                  <PopoverBody id={`popover-${randomId}-body`}>{bodyContent}</PopoverBody>
+                </PopoverContent>
+              </div>
+            </FocusTrap>
+          )
+        }
+      </GenerateId>
+    );
+    const handleEvents = isVisible === null;
+    const shouldHideOnClick = () => {
+      if (handleEvents) {
+        if (hideOnOutsideClick === true) {
+          return true;
+        }
+        return 'toggle';
+      }
+      return false;
+    };
+    return (
+      <Tippy
+        onCreate={this.storeTippyInstance}
+        size={size}
+        zIndex={zIndex}
+        appendTo={appendTo}
+        content={content}
+        lazy
+        trigger={handleEvents ? 'click' : 'manual'}
+        isVisible={isVisible}
+        hideOnClick={shouldHideOnClick()}
```

Comment:
do you really want to call the function here?

Reply:
Yes, it basically maps the prop hideOnOutsideClick with values of `true` or `false` to `true` and `'toggle'`